helix-core/src/main/java/org/apache/helix/tools/ClusterSetup.java

```diff
@@ -176,6 +179,30 @@ public void close() {
     }
   }
 
+  public void addCluster(String clusterName, boolean overwritePrevious, boolean isCloudEnabled)
```

Comment:
Thinking this thoroughly, I believe we need more parameters than "isCloudEnabled". cloudID is dynamically generated every time, and we do not have default value for it. With that said, it makes more sense for users to be able to input "cloudProvider" and "cloudId" when the cloud is "enabled". We should assume that we always have "Azure" as default.

Reply:
Just my 2 cents: to me, passing a boolean param to a method to determine the behavior is kind of code smell.
See https://martinfowler.com/bliki/FlagArgument.html
Can we have a better design for this?

Reply:
Fixed. Also tight now we have more input parameter.